.travis.yml

```diff
@@ -1,28 +1,87 @@
 language: php
 
-php:
-    - 7.0
-    - 7.1
-    - 7.2
-
 cache:
-    directories:
-        - $HOME/.composer/cache/files
+  directories:
+    - $HOME/.composer/cache/files
 
-before_install:
-    - phpenv config-rm xdebug.ini
-    - composer validate
+stages:
+  - stan
+  - test
 
-install:
-    - composer update --prefer-dist --prefer-stable --no-interaction
+jobs:
+  include:
+    - stage: Stan
+
+      php: 7.2
+
+      install:
+        - composer install
+
+      script:
+        - vendor/bin/phpstan analyse -l max -c phpstan.neon src tests
+
+    - &TEST
+
+      stage: Test
+
+      php: 7.0
+
+      env: WITH_LOWEST=true
+
+      install:
+        - if [[ "$WITH_LOWEST" == "true" ]]; then composer update --prefer-lowest; fi
+        - if [[ "$WITH_LOCKED" == "true" ]]; then composer install; fi
+        - if [[ "$WITH_HIGHEST" == "true" ]]; then composer update; fi
+
+      script:
+        - vendor/bin/phpunit
+
+    - <<: *TEST
+
+      php: 7.0
+
+      env: WITH_LOCKED=true
+
+    - <<: *TEST
+
+      php: 7.0
+
+      env: WITH_HIGHEST=true
+
+    - <<: *TEST
+
+      php: 7.1
+
+      env: WITH_LOWEST=true
+
+    - <<: *TEST
+
+      php: 7.1
+
+      env: WITH_LOCKED=true
+
+    - <<: *TEST
+
+      php: 7.1
+
+      env: WITH_HIGHEST=true
+
+    - <<: *TEST
+
+      php: 7.2
+
+      env: WITH_LOWEST=true
+
+    - <<: *TEST
+
+      php: 7.2
+
+      env: WITH_LOCKED=true
+
+    - <<: *TEST
+
+      php: 7.2
+
+      env: WITH_HIGHEST=true
 
-script:
-    - vendor/bin/phpstan analyse -l max -c phpstan.neon src tests
-    - vendor/bin/phpunit
 
-jobs:
-    include:
-        - stage: Test
-            php: 7.0
```

Comment:
This is what I mean in #17, the indentation is just off.

Reply:
Thanks, this makes it more obvious.